docs/maven-plugin.md

```diff
@@ -0,0 +1,174 @@
+# Usage
+
+Add the following profiles to your pom.xml:
+
+```
+  <profile>
+    <id>check</id>
+    <build>
+      <pluginManagement>
+        <plugins>
+          <plugin>
+            <groupId>org.openhab.tools</groupId>
+            <artifactId>static-code-analysis</artifactId>
+            <version>${sat.version}</version>
+            <executions>
+              <execution>
+                <phase>verify</phase>
+                <goals>
+                  <goal>checkstyle</goal>
+                  <goal>pmd</goal>
+                  <goal>findbugs</goal>
+                  <goal>report</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </pluginManagement>
+    </build>
+  </profile>
+  <profile>
+    <id>check-bundles</id>
+    <activation>
+      <file>
+        <exists>META-INF/MANIFEST.MF</exists>
+      </file>
+    </activation>
+    <build>
+      <plugins>
+        <plugin>
+            <groupId>org.openhab.tools</groupId>
+            <artifactId>static-code-analysis</artifactId>
+        </plugin>
+      </plugins>
+    </build>
+  </profile>
+```
+
+ Execute `mvn clean install -P check` from the root of your project.
+
+ Reports are generated for each module individually and can be found in the `target/code-analysis` directory. The merged report can be found in the root target directory.
+
+ The build will fail if a problem with high priority is found by some of the Maven plugins for PMD, Checkstyle and FindBugs. Each of the plugins has its own way to prioritize the detected problems:
+
+ - for PMD - the build will fail when a rule with Priority "1" is found;
+ - for Checkstyle - a rule with severity="Error";
+ - for Findbugs - any Matcher with Rank between 1 and 4.
+
+## Maven plugin goals and parameters
+

```

Comment:
I barely got the idea of this section - to outline some additional configuration that can be done before running the tool. But as I have some knowledge on the whole concept I guess that it will be harder for a newcomer to realize the same as me. so can you add a slight introduction fir this section as well ?